crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1622,6 +1625,30 @@ impl AuthorityPerEpochStore {
         self.load_deferred_transactions(batch, min, max)
     }
 
+    fn load_and_process_deferred_transactions_for_randomness(
+        &self,
+        batch: &mut DBBatch,
+        previously_deferred_tx_digests: &mut HashMap<TransactionDigest, DeferralKey>,
+        sequenced_randomness_transactions: &mut Vec<VerifiedSequencedConsensusTransaction>,
+    ) -> SuiResult {
+        let deferred_randomness_txs = self.load_deferred_transactions_for_randomness(batch)?;
+        previously_deferred_tx_digests.extend(deferred_randomness_txs.iter().flat_map(
+            |(deferral_key, txs)| {
+                txs.iter().map(|tx| match tx.0.transaction.key() {
+                    SequencedConsensusTransactionKey::External(
+                        ConsensusTransactionKey::Certificate(digest),
+                    ) => (digest, *deferral_key),
+                    _ => {
+                        panic!("deferred randomness transaction was not a user certificate: {tx:?}")
+                    }
+                })
+            },
+        ));
+        sequenced_randomness_transactions
```

Comment:
I'm a bit confused about the difference between the transactions being processed here and the ones from the code below - 
`for tx in deferred_txs
            .into_iter()
            .flat_map(|(_, txs)| txs.into_iter())
        `
1. what is the difference between them? 
2. say we generate randomness in the current commit - what will be the order of sequenced transactions from the current commit & from the previous commit (say assuming we generate randomness every 4 commits)?

Reply:
1. code below is for deferral due to congestion, not randomness

2. it will add tx back into the sequence ordered first by commit round from which they were deferred, and then preserving order of tx from consensus within that round.

I do think this interacted incorrectly with randomness tx deferred due to congestion, which would have been added to the front of the list the way the code is currently ordered. I swapped it around so that they go at the start of the current commit's randomness-using tx, same as it works for non-randomness tx